pool/web/view/page/page_admin_contact.ml

```diff
@@ -386,33 +386,75 @@ let experiment_history Pool_context.{ language; _ } contact experiments query =
 ;;
 
 let experiment_history_modal
-      { Pool_context.language; _ }
+      { Pool_context.csrf; language; _ }
       (experiment : Experiment.t)
       assignments
   =
   let open Pool_common in
   let title (_ : Language.t) = Experiment.(Title.value experiment.title) in
   let html =
     let thead =
-      Pool_message.Field.[ Start; NoShow; Participant ]
+      Pool_message.Field.[ Start; NoShow; Participated ]
       |> CCList.map CCFun.(Utils.field_to_string_capitalized language %> txt)
     in
+    let thead = thead @ [ txt "" ] in
     let to_icon value =
       CCOption.map_or ~default:(txt "") CCFun.(value %> Component.Icon.bool_to_icon)
     in
     let rows =
       let open Assignment in
+      let open Session in
       assignments
-      |> CCList.map (fun (session, { no_show; participated; _ }) ->
-        let start = Session.start_end_with_duration_human session |> txt in
+      |> CCList.map (fun (session, { id; no_show; participated; _ }) ->
+        let start = start_end_with_duration_human session |> txt in
         let start =
-          if CCOption.is_some session.Session.follow_up_to
+          if CCOption.is_some session.follow_up_to
           then div ~a:[ a_class [ "inset"; "left" ] ] [ start ]
           else start
         in
-        [ start; to_icon NoShow.value no_show; to_icon Participated.value participated ])
+        let mark_as_deleted =
+          match session.closed_at with
+          | Some _ -> txt ""
+          | None ->
+            let action =
+              Http_utils.Url.Admin.assignment_path
+                experiment.Experiment.id
+                session.id
+                ~id
+                ~suffix:"mark-as-deleted"
+                ()
+              |> Sihl.Web.externalize_path
+            in
+            form
+              ~a:
+                [ a_method `Post
+                ; a_action action
+                ; a_user_data
+                    "confirmable"
+                    (Utils.confirmable_to_string
+                       language
+                       Pool_common.I18n.MarkAssignmentWithFollowUpsAsDeleted)
+                ]
+              Component.Input.
+                [ csrf_element csrf ()
+                ; submit_icon
+                    ~attributes:
+                      [ a_title
+                          (Utils.control_to_string
+                             language
+                             Pool_message.Control.MarkAsDeleted)
+                      ]
+                    ~classnames:[ "error" ]
+                    Icon.TrashOutline
+                ]
+        in
+        [ start
+        ; to_icon NoShow.value no_show
+        ; to_icon Participated.value participated
+        ; mark_as_deleted
```

Comment:
Additionally: add possibility to "Cancel" the assignment (prohibit from resignup)